docs/adr/0004-leave-implementation-of-synchronous-dispatcher-to-users-of-the-library.md

```diff
@@ -8,12 +8,27 @@ Accepted
 
 ## Context
 
-The issue motivating this decision, and any context that influences or constrains the decision.
+The library at the early stages had a functionality of dispatching in-memory events.
+
+A user of the library could plug-in listeners to particular types of events (also - catch-all listener was supported).
+When events were published via `EventStore` with configured listeners, the latter were called synchronously one after another. This happened right after events were persisted.
+
+This delivered a very basic mechanism of subscribing to events from e.g. other code modules, similar to e.g. Django signals.
+
+However, in-memory events implementation failed to conform to Liskov's Substitute Principle, because the handling and consequences of the erorrs are different for SQL-based backend and Event Store DB.
```

Comment:
```suggestion
However, in-memory events implementation failed to conform to Liskov's Substitute Principle, because the handling and consequences of the errors are different for SQL-based backend and Event Store DB.
```